tests/unit/jira/test_utils.py

```diff
@@ -0,0 +1,68 @@
+from textwrap import dedent
+
+import pytest
+
+from jbi.jira.utils import markdown_to_jira
+
+
+def test_markdown_to_jira():
+    markdown = dedent(
+        """
+    Mixed nested lists
+
+    * a
+    * bulleted
+      - with
+      - nested
+        1. nested-nested
+      - numbered
+    * list
+
+    this was `inline` value ``that`` is turned into ```monospace``` tag.
+
+    this sentence __has__ **bold** and _has_ *italic*.
+
+    this was ~~wrong~~.
+    """
+    ).lstrip()
+
+    jira = dedent(
+        """
+    Mixed nested lists
+
+    * a
+    * bulleted
+    ** with
+    ** nested
+    **# nested-nested
+    ** numbered
+    * list
+
+    this was {{inline}} value {{that}} is turned into {{monospace}} tag.
+
+    this sentence *has* *bold* and _has_ _italic_.
+
+    this was -wrong-.
+    """
+    ).strip()
+
+    assert markdown_to_jira(markdown) == jira
+
+
+def test_markdown_to_jira_with_malformed_input():
+    assert markdown_to_jira("[link|http://noend") == "\\[link|http://noend"
```

Comment:
Nice that Pandoc handles this no problem, but wish it'd return an empty string rather than the mangled output `\\[link|http://noend`. But overall, we're in a much better place than when we started. Maybe worth spinning an issue up for this for polish, but good enough for this PR.

Reply:
I'm not sure about your suggestion, in markup languages when you make a syntax mistake, it shows the markup "as is" [example[link] or [example](link

Reply:
Hmm I guess you're right. This is what I'd want though:

```
>>> markdown_to_jira("The line is [too long|https://www.example.com]", max_len=20)
"This line is ..."
```

I think this would rely on us converting the entire comment to the new markup, then looking for a good place to truncate after the fact. 

I don't know if this sort of thing even makes sense, but in my head, Pandoc would consider `[too long|https://www.example.com]` as an entire "block" or "unit of markup" (as it would `**bold**` or `## Heading`), and then split on a block, rather than on a space in the middle of a "block".

Reply:
Ok, I understand what you mean now.

Well, we split on space so the result would be:

```
>>> markdown_to_jira("The line is [too long|https://www.example.com]", max_len=20)
"This line is [too"
```

We could also truncate on `max_length - 3` and add `...`  or something more explicit like `max_length - len("[truncated...]")`


> I think this would rely on us converting the entire comment to the new markup, then looking for a good place to truncate after the fact.

Well, it would require having access to the semantics of characters in their context, not just the output of the new markup